core/src/main/scala/org/apache/spark/h2o/utils/H2OContextUtils.scala

```diff
@@ -51,6 +54,60 @@ private[spark] trait H2OContextUtils extends Logging {
     }
   }
 
+  private def isTcpPortAvailable(port: Int): Boolean = {
+    try {
+      val serverSocket = new ServerSocket()
+
+      serverSocket.setReuseAddress(false);
+      serverSocket.bind(new InetSocketAddress(InetAddress.getByName(SparkEnv.get.blockManager.blockManagerId.host), port), 1)
+      serverSocket.close()
+      true
+    } catch {
+      case _: Exception => false
+    } finally {
+    }
+  }
+
+  private def findNextFreeFlowPort(conf: H2OConf): Int = {
+    if (conf.clientWebPort == -1) {
+      var port = conf.clientBasePort
```

Comment:
Wasn't your intention to implement something like this?
```
var port = conf.clientBasePort + conf.internalPortOffset
while (!isTcpPortAvailable(port)) {
    port = port + 1
}
port

Reply:
Thanks for checking!

I had a second look, but still think that the code bellow is the intended one. In H2O, the `internalPortOffset` specifies the offset between the API port (54321) and internal port(54320). Even though we don't run any H2O client here, I wanted to be consistent in port selection. Does it make sense?

Reply:
It makes sense what you wrote, but still don't understand why you use `offset` for  a jump if a port is already occupied `port = port + offset + 1`. Why isn't the next one tried if it's free or not?

Reply:
I see what you're trying to say now! Yes, that is a good point!

Reply:
After thinking more about it, I simplified it. We do not need to use offset here as we do not have any internal communication port. Therefore we just try all the port starting from the base port